ansible/roles/test/tasks/snmp/memory.yml

```diff
@@ -0,0 +1,105 @@
+# Test SNMP Memory Utilization
+#
+#  - Check the difference percentage of Total Memory between SNMP and shell is 0
+#  - Check the difference percentage of Total Free Memory between SNMP and shell small than 3% when:
+#    * Check free memory before stress test
+#    * Check free memory in stress test
+#    * Check free memory after stress test
+#  - Check memory leak issue, the difference percentage between Total Free Memory before stress test and after should small than 3%
+#
+# Requires:  Ansible v2.0.0.2-1
+#
+# Usage:
+#
+#  ansible-playbook test_sonic.yml -i inventory --limit switch2 --become --tags snmp_memory -vvvvvv
+#
+
+- block:
+    - name: Copy Memory stress script to the switch
+      copy: src={{ item.src }} dest={{ item.dest }}
+      with_items:
+        - { src: 'roles/test/tasks/snmp/memory.py', dest: '/tmp/memory.py' }
+
+    - name: Gathering basic snmp facts about the device
+      snmp_facts: host={{ ansible_host }} version=v2c community={{ snmp_rocommunity }}
+      connection: local
+
+    - name: Check total memory via shell
+      shell: grep MemTotal /proc/meminfo | awk '{print $2}'
+      register: shell_total_memory
+      become: yes
+
+    - name: 'Validating SNMP total memory matches shell "/proc/meminfo" result'
+      assert:
+        that: "{{ (ansible_sysTotalMemery|int - shell_total_memory.stdout|int)|abs }} == 0"
+
+    - name: Gathering basic snmp facts about the device
+      snmp_facts: host={{ ansible_host }} version=v2c community={{ snmp_rocommunity }}
+      connection: local
+
+    - name: Check total free memory via shell
+      shell: grep MemFree /proc/meminfo | awk '{print $2}'
+      register: shell_total_free_memory
+      become: yes
+
+    - name: Validating SNMP total free memory matches shell result before stress test
+      assert:
+        that: "{{ '%.6f'|format((ansible_sysTotalFreeMemery|int - shell_total_free_memory.stdout|int)|abs / ansible_sysTotalFreeMemery|int) }} <= 0.03"
+
+    - set_fact: free_memery_before_test="{{ ansible_sysTotalFreeMemery }}"
+
+    - set_fact: test_momory="{{ ((ansible_sysTotalFreeMemery - 512000) / 1024)|int }}"
```

Comment:
why minus 512M memory? what is the purpose of memory stress test? is it just to verify the snmp can pick the memory consumption changes or there is some other goal? if only the first goal, then we just need to fill a few hundreds MB and check. We do not need to fill all the available memory.

Reply:
This test will also check the SNMP process is still functional or not in suck low free memory condition.
It's enough for my box that only left 512M for system free memory.
Please let me know if any.

Reply:
my request here is that can we skip this, if the sysTotalFreeMemory is already less then 512000? As above, can we make this as a tunable parameter, you can make 512M as the default value.